packages/components/src/utils/element-rect.ts

```diff
@@ -79,9 +80,11 @@ export function getElementOffsetRect(
 	if ( rect.width === 0 || rect.height === 0 ) {
 		return;
 	}
+	const offsetParent = element.offsetParent;
 	const offsetParentRect =
-		element.offsetParent?.getBoundingClientRect() ??
-		NULL_ELEMENT_OFFSET_RECT;
+		offsetParent?.getBoundingClientRect() ?? NULL_ELEMENT_OFFSET_RECT;
```

Comment:
I wonder if we can improve runtime performance by "caching" offset parents and scroll positions, and only update them as necessaily

Reply:
This computation only happens on target changes or resizes. Caching here would defeat the purpose since every time we run the computation we have good reason to think it has potentially changed, and we can't know it has not without accessing this information in the first place.

I'd be surprised if this had any performance issues, the computation rarely runs (on tab change or when the active tab changes size which is now almost impossible since it doesn't even wrap anymore.